src/Components/Components/test/Lifetime/ComponentApplicationLifetimeTest.cs

```diff
@@ -99,18 +101,18 @@ public async Task PersistStateAsync_InvokesPauseCallbacksDuringPersist()
         public async Task PersistStateAsync_FiresCallbacksInParallel()
         {
             // Arrange
-            var state = new Dictionary<string, byte[]>();
+            var state = new Dictionary<string, ReadOnlySequence<byte>>();
             var store = new TestStore(state);
-            var lifetime = new ComponentApplicationLifetime(NullLogger<ComponentApplicationLifetime>.Instance);
+            var lifetime = new ComponentStatePersistenceManager(NullLogger<ComponentStatePersistenceManager>.Instance);
             var renderer = new TestRenderer();
 
             var sequence = new List<int> { };
 
             var tcs = new TaskCompletionSource();
             var tcs2 = new TaskCompletionSource();
 
-            lifetime.State.OnPersisting += async () => { sequence.Add(1); await tcs.Task; sequence.Add(3); };
-            lifetime.State.OnPersisting += async () => { sequence.Add(2); await tcs2.Task; sequence.Add(4); };
+            lifetime.State.RegisterOnPersisting(async () => { sequence.Add(1); await tcs.Task; sequence.Add(3); });
+            lifetime.State.RegisterOnPersisting(async () => { sequence.Add(2); await tcs2.Task; sequence.Add(4); });
```

Comment:
I might be missing something, but do any of the tests cover checking that the subscription can be successfully disposed? It looks as if they all register but don't unregister.